### PR TITLE
Update the babel-compiler package use to Babel 7, like the command-line tool.

### PR DESCRIPTION
**Pull request by benjamn**
Since Meteor 1.6, we've been quietly beta-testing Babel 7 by using it to compile [the `meteor` command-line tool codebase](https://github.com/meteor/meteor/tree/devel/tools). This [dog-fooding](https://en.wikipedia.org/wiki/Eating_your_own_dog_food) was vital when we first introduced Babel compilation in Meteor 1.2, and it later helped us gain confidence in the upgrade from Babel 5 to Babel 6. Now it's time to make the leap to Babel 7!

There are a number of important changes hidden behind the abstraction of [`meteor-babel`](https://github.com/meteor/babel/commits/master), but the most disruptive change that will be visible to application developers is that the [`babel-runtime`](https://www.npmjs.com/package/babel-runtime) package is now called [`@babel/runtime`](https://www.npmjs.com/package/@babel/runtime). Since this package is typically installed in the application `node_modules` directory, developers will likely need to run `meteor npm install @babel/runtime` and/or edit their `package.json` files.

Probably the next most disruptive change is that the Meteor `babel-compiler` package has jumped a major version, from 6.24.7 to 7.0.0, which may require packages that depend directly on `babel-compiler` to adjust their version constraints.

There are a few other notable changes, but I think it will be easier if I comment on the commits themselves, below.

### CODE REVIEW

File: packages/non-core/coffeescript-compiler/package.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ Npm.depends({
 });
 
 Package.onUse(function (api) {
-  api.use('babel-compiler@6.19.4');
+  api.use('babel-compiler@6.19.4||7.0.0');
```
Comment (benjamn) [inline packages/non-core/coffeescript-compiler/package.js]: @GeoffreyBooth Good news: this syntax actually works!

File: packages/ecmascript/transpilation-tests.js [hunk 2]
```diff
@@ -62,7 +66,7 @@ class Foo {}
 class Bar extends Foo {}
 `);
 
-  test.isTrue(contains(output, 'helpers/inherits'));
+  test.isTrue(contains(output, 'helpers/builtin/inherits'));
```
Comment (benjamn) [inline packages/ecmascript/transpilation-tests.js]: This change was necessary because of https://github.com/meteor/babel/commit/242538dea13dd4600e2ba2fc1d3bd7d242e35304, which makes [`@babel/plugin-tranform-runtime`](https://www.npmjs.com/package/@babel/plugin-transform-runtime) generate imports for `@babel/runtime/helpers/builtin/whatever` instead of `@babel/runtime/helpers/whatever`. The `helpers/builtin/*` helpers are important because they do not import any polyfills from `core-js`, and instead simply assume global APIs like `Symbol` are natively supported or have been properly polyfilled. Since Meteor takes care to provide appropriate polyfills, especially with #9439, it's important to let other parts of the framework import `core-js` polyfills, rather than leaving that job to a generic tool like Babel.

File: packages/tinytest/tinytest.js [hunk 3]
```diff
@@ -70,8 +70,9 @@ export class TestCaseResults {
         var frame = stack[i];
         // Heuristic: use the OUTERMOST line which is in a :tests.js
         // file (this is less likely to be a test helper function).
-        if (frame.getFileName().match(/:tests\.js/)) {
-          doc.filename = frame.getFileName();
+        const fileName = frame.getFileName();
+        if (fileName && fileName.match(/:tests\.js/)) {
+          doc.filename = fileName;
```
Comment (benjamn) [inline packages/tinytest/tinytest.js]: I am not sure why this change became necessary after upgrading Babel, but somehow `frame.getFileName()` started returning `null` in some cases, so I decided to make this code more defensive.

File: packages/non-core/coffeescript-compiler/package.js (hunk 1 above)
Comment (GeoffreyBooth) [reply packages/non-core/coffeescript-compiler/package.js]: This is something, at least. It lets me as a package developer write something like `api.use('coffeescript@1||2');` (I assume that works, and not just exact `1.12.7||2.0.3`?)

It doesn’t really solve the issue of #9244, though, because to take advantage of this syntax you still need to fork a package. There still needs to be some way for end users to override the dependencies a package developer has set, if they haven’t thought to use this syntax or they picked versions that don’t work with other packages the user might want to install. But this is progress.

Comment (benjamn) [reply packages/non-core/coffeescript-compiler/package.js]: Totally agree!
